apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -233,7 +244,7 @@ defmodule Aecore.Chain.Worker do
     chain_states =
       case Persistence.get_all_accounts_chain_states() do
         chain_states when chain_states == %{} -> state.chain_states
-        chain_states -> %{top_hash => chain_states}
+        chain_states -> %{top_hash => chain_states["chain_state"]}
```

Comment:
looks weird, can you explain why this call is needed? Is `chain_states` the correct semantical name still? If so why does it have a `"chain_state"` item?

Reply:
This is because before the chain_state was actually only holding the accounts, but now we want the chainstate to hold other {key, value} pairs as well. We typed it this way so that we save the whole chain state in the db, not only the accounts: key values, but all the other pairs that will come in the future as well. And then to get them from the db, you'll only need to create a extraction function in the persistence module

Reply:
what is included in chain_states?

Reply:
@gspasov ?

Reply:
At the moment the chain_state holds only the ket `:accounts` with its map of values of all the accounts, but later it will hold all the specific transaction states as well in there.